### PR TITLE
Kati-73 Health tracker function

### PR DESCRIPTION
**Pull request by toannguyen71195**

### CODE REVIEW

File: app/src/main/java/com/kms/cura/model/UserModel.java [hunk 1]
```diff
@@ -62,5 +62,21 @@ public void userLogin(final UserEntity entity) {
...
+        StringRequest stringRequest = RequestUtils.createRequest(builder.toString(),
+                Request.Method.POST, EntityToJsonConverter.convertEntityToJson(entity).toString(),
+                response);
+        VolleyHelper.getInstance().addToRequestQueue(stringRequest, tag_string_req);
+        while (!response.isGotResponse());
+        if(!response.isResponseError()) {
+            healthEntities = response.getEntityList();
+            return;
+        }
+        throw new Exception(response.getError()); */
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/UserModel.java]: Remove this comment block

File: app/src/main/java/com/kms/cura/view/fragment/HealthTrackerFragment.java [hunk 2]
```diff
@@ -105,4 +88,50 @@ private void resetAdapter(ArrayList<String> currentHealth, ArrayList<String> pas
...
+        for (int i = 0; i < list.size(); i++) {
+            names.add(list.get(i).getName());
+        }
+        return names;
+    }
+
+    private ArrayList<HealthEntity> getCurrentHealth(List<HealthEntity> list) {
+        ArrayList<HealthEntity> current = new ArrayList<>();
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getEndDate() == null) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/HealthTrackerFragment.java]: Should put getEndDate() == null inside a method called isPastHealth inside HealthEntity

File: app/src/main/java/com/kms/cura/view/fragment/HealthTrackerFragment.java [hunk 3]
```diff
@@ -105,4 +88,50 @@ private void resetAdapter(ArrayList<String> currentHealth, ArrayList<String> pas
...
+                current.add(list.get(i));
+            }
+        }
+        return current;
+    }
+
+    private ArrayList<HealthEntity> getPastHealth(List<HealthEntity> list) {
+        ArrayList<HealthEntity> past = new ArrayList<>();
+        for (int i = 0; i < list.size(); i++) {
+            if (list.get(i).getEndDate() != null) {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/HealthTrackerFragment.java]: After that then use !(list.get(i).isPastHealth()) for this

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+	private static final String SYMPTOM_NAME = "symptom_name";
+
+	public PatientHealthDatabaseHelper() throws ClassNotFoundException, SQLException {
+		super();
+	}
+
+	@Override
+	protected HealthEntity getEntityFromResultSet(ResultSet resultSet)
+			throws SQLException, ClassNotFoundException {
+		// TODO Auto-generated method stub
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Remove TODO comment

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+		builder.append(condition_id);
+		builder.append(", ");
+		builder.append(startDate.toString());
+		builder.append(", ");
+		builder.append(endDate.toString());
+		builder.append(")");
+		try {
+			stmt = con.prepareStatement(builder.toString());
+			stmt.executeUpdate();
+			con.commit();
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Insert (Update) methods need rollback step if failed

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+		builder.append(patient_id);
+		builder.append(", ");
+		builder.append(condition_id);
+		builder.append(", ");
+		builder.append(startDate);
+		builder.append(")");
+		try {
+			stmt = con.prepareStatement(builder.toString());
+			stmt.executeUpdate();
+			con.commit();
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Same here, need rollback

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+		builder.append(symptom_id);
+		builder.append(", ");
+		builder.append(startDate);
+		builder.append(", ");
+		builder.append(endDate);
+		builder.append(")");
+		try {
+			stmt = con.prepareStatement(builder.toString());
+			stmt.executeUpdate();
+			con.commit();
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Same here, need rollback

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+		builder.append(patient_id);
+		builder.append(", ");
+		builder.append(symptom_id);
+		builder.append(", ");
+		builder.append(startDate);
+		builder.append(")");
+		try {
+			stmt = con.prepareStatement(builder.toString());
+			stmt.executeUpdate();
+			con.commit();
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Same here, need rollback

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+			List<HealthEntity> pHealthEntities = queryHealthByPatientID(patient_id);
+			return queryHealthByPatientID(patient_id).get(pHealthEntities.size() - 1);
+		} finally {
+			stmt.close();
+		}
+	}
+
+	public HealthEntity insertPatient_SymptomPresent(String patient_id, String symptom_id, Date startDate)
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientHealthDatabaseHelper.java]: Do we need separated methods for Past and Present Symptom ( or Condition)

File: com.kms.cura.dal/src/com/kms/cura/dal/database/PatientUserDatabaseHelper.java [hunk 10]
```diff
@@ -110,16 +111,32 @@ protected UserEntity getEntityFromResultSet(ResultSet resultSet) throws SQLExcep
...
+		if (userEntity != null) {
+			return new PatientUserEntity(resultSet.getString(PatientColumn.USER_ID.getColumnName()),
+					resultSet.getString(PatientColumn.NAME.getColumnName()), userEntity.getEmail(),
+					userEntity.getPassword(), resultSet.getString(PatientColumn.GENDER.getColumnName()),
+					resultSet.getDate(PatientColumn.BIRTH.getColumnName()),
+					resultSet.getString(PatientColumn.LOCATION.getColumnName()),
+					resultSet.getString(PatientColumn.INSURANCE.getColumnName()),
+					resultSet.getString(PatientColumn.HEALTH_CONCERN.getColumnName()));
+		}
+		return null;*/
```
Comment (hieumai) [inline com.kms.cura.dal/src/com/kms/cura/dal/database/PatientUserDatabaseHelper.java]: Remove comment block

File: com.kms.cura_server/src/com/kms/cura_server/PatientHealthAPI.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+import com.kms.cura.dal.mapping.PatientHealthColumn;
+import com.kms.cura.dal.mapping.UserColumn;
+import com.kms.cura.dal.user.PatientUserDAL;
+import com.kms.cura.entity.Entity;
+import com.kms.cura.entity.HealthEntity;
+import com.kms.cura_server.response.APIResponse;
+import com.kms.cura_server.response.PatientHealthResponse;
+
+@Path("/patient_health")
+public class PatientHealthAPI {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/PatientHealthAPI.java]: Remove this class and move all the methods inside Patient

File: app/src/main/java/com/kms/cura/view/activity/LoginActivity.java [hunk 12]
```diff
@@ -196,4 +200,14 @@ protected void onResume() {
...
     }
 
+    private void showProgressDialog() {
+        pDialog = new ProgressDialog(this);
+        pDialog.setMessage("Loading...");
+        pDialog.setCancelable(false);
+        pDialog.show();
+    }
+
+    private void hideProgressDialog() {
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/activity/LoginActivity.java]: remove showProgressDialog and hideProgressDialog for not using

File: com.kms.cura_server/src/com/kms/cura_server/UserAPI.java [hunk 13]
```diff
@@ -122,5 +128,29 @@ public String userLogin(String jsonData){
 		}
 
 	}
+	
+	@GET
+	@Path("/getAllPatientHealth")
+	public String getAllPatientHealth() {
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/UserAPI.java]: Do we need this API ?

**Comment by hieumai** [comment]
OK from me. @huuduy216  and @vtnlinh95  please start verifying this

**Comment by vtnlinh95** [comment]
@toannguyen71195 stop working when login . Please check

**Comment by huuduy216** [comment]
verified ok

**Pull request merged**